src/sdcard/mod.rs

```diff
@@ -206,89 +417,73 @@ where
                     Ok(R1_IDLE_STATE) => {
                         break;
                     }
-                    Ok(r) => {
+                    Ok(_r) => {
                         // Try again
-                        warn!("Got response: {:x}, trying again..", r);
+                        warn!("Got response: {:x}, trying again..", _r);
+                        attempts -= 1;
```

Comment:
`delay` already contains a retry count. With this change, that becomes entirely redundant.
The retry count of `delay` is much higher though, 32000.

Reply:
Fair point, well made. I've changed it to use the Delay max retry count, and set that to a more reasonable max 512 retries.

Reply:
512 retries with a 10µs delay each are ~5ms. I guess that's enough in most cases, but I'm not sure if it's sufficient in general: For example, the same retry count is used when waiting for the response to a write command. What if a write command needs to perform some defragmentation due to the card's write leveling algorithm?
I didn't check the spec to see if there are maximum delays specified. But the old value of 32000 retries / 0.3s, while rather long, isn't necessarily wrong. Without knowing why the value was chosen that high in the first place, I'd be worried that it might be needed in some cases. (Chesterton's fence...)

So it might be better to keep the old value as a default, and use `Delay::new( lower_value )` in cases where a shorter timeout is desired?